packages/scoutgame-ui/src/components/accounts/components/TelegramConnect.tsx

```diff
@@ -66,67 +51,128 @@ export function TelegramConnect({ user }: { user: UserWithAccountsDetails }) {
     }
   );
 
-  useEffect(() => {
-    const script = document.createElement('script');
-    script.src = 'https://telegram.org/js/telegram-widget.js?22';
-    script.async = true;
-    document.getElementById('telegram-login-container')?.appendChild(script);
-  }, []);
+  const { executeAsync: verifyTelegramToken } = useAction(verifyTelegramTokenAction, {
+    onSuccess({ data }) {
+      const telegramAccount = data?.telegramUser;
+      if (telegramAccount && 'id' in telegramAccount) {
+        setAuthData(telegramAccount);
+        connectTelegramAccount(telegramAccount);
+      } else {
+        log.error('Invalid Telegram account', { telegramAccount });
+        setConnectionError('Invalid Telegram account');
+      }
+      setQrCode(null);
+    },
+    onError({ error }) {
+      if (error.serverError?.message?.toLowerCase().includes('session password')) {
+        setIsSessionPasswordNeeded(true);
+      }
+    }
+  });
+
+  const { executeAsync: removeTelegramClient } = useAction(removeTelegramClientAction);
+
+  const { executeAsync: generateTelegramQrCode, isExecuting: isGeneratingQrCode } = useAction(
+    generateTelegramQrCodeAction,
+    {
+      onSuccess: (data) => {
+        if (data.data?.qrCodeImage) {
+          setQrCode(data.data.qrCodeImage);
```

Comment:
I don't think you need a state for qrCode since you can use the result?.data?.qrCodeImage.
And when you verifyTelegramToken() -> onSuccess you can use `reset` to reset the qrCodeImage.

Reply:
Nvm. I see that this func needs to be after the above one since it uses `verifyTelegramToken`. That means you cant `reset` qrCodeImage from above.